pages/expenses/details/[id].js

```diff
@@ -1,25 +1,43 @@
 import ExpenseDetail from "../../../components/expenses/expenseDetail";
 import styled from "styled-components";
-import { getExpenseById } from "../../../services/expenseServices";
 import Link from "next/link";
 import EditButton from "../../../components/buttons/editButton";
+import useSWR from "swr";
+import lottie from "lottie-web";
+import { useEffect, useRef } from "react";
 
-export async function getServerSideProps(ctx) {
-  const { id } = ctx.params;
-  const expense = await getExpenseById(id);
+const fetcher = async () => {
+  const pathArray = window.location.pathname.split("/");
+  const id = pathArray[3];
+  const response = await fetch(`/api/expenses/${id}`);
+  const data = await response.json();
+  return data;
+};
 
-  return {
-    props: { expense },
-  };
-}
+export default function ExpenseDetails() {
+  const container = useRef(null);
+
+  useEffect(() => {
+    lottie.loadAnimation({
+      container: container.current,
+      render: "svg",
+      loop: true,
+      autoplay: true,
+      animationData: require("../../../public/loading_animation.json"),
+    });
+  }, []);
```

Comment:
The loading animation is shown only once.
If you want your animation to indicate a loading state, you'd need to show it every time the component is rendering while fetching data.